cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -886,12 +890,27 @@ func (p *BicepProvider) loadParameters(
 		return nil, fmt.Errorf("substituting environment variables inside parameter file: %w", err)
 	}
 
-	if cmdsubst.ContainsCommandInvocation(replaced, cmdsubst.SecretOrRandomPasswordCommandName) {
-		cmdExecutor := cmdsubst.NewSecretOrRandomPasswordExecutor(p.azCli, p.env.GetSubscriptionId())
+	if cmdsubst.ContainsCommandInvocation(replaced, cmdsubst.SecretOrRandomPasswordCommandName) ||
+		cmdsubst.ContainsCommandInvocation(replaced, cmdsubst.ContainerAppImageCommandName) {
+
+		projectConfig, err := project.Load(ctx, filepath.Join(p.projectPath, azdcontext.ProjectFileName))
+		if err != nil {
+			return nil, fmt.Errorf("reading project: %w", err)
+		}
+
+		cmdExecutor := cmdsubst.Join(
+			cmdsubst.NewSecretOrRandomPasswordExecutor(p.azCli, p.env.GetSubscriptionId()),
+			cmdsubst.NewContainerAppImageExecutor(
+				project.NewResourceManager(p.env, p.azCli),
+				p.containerAppService,
+				p.env.GetSubscriptionId(),
+				projectConfig.Services,
+			))
```

Comment:
Yeah - that makes sense.  I think it would also save us some work if we injected whatever component does the command evaluation into the provider instead of creating it here.  Fewer dependencies to flow across the boundary and better encapsulation.